src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -103,6 +135,33 @@ public void Add(X509Certificate2 certificate)
             _storePal.Add(certificate.Pal);
         }
 
+        public void AddRange(X509Certificate2Collection certificates)
+        {
+            if (certificates == null)
+                throw new ArgumentNullException(nameof(certificates));
+
+            int i = 0;
+            try
+            {
+                foreach (X509Certificate2 certificate in certificates)
+                {
+                    Add(certificate);
+                    i++;
+                }
+            }
+            catch
+            {
+                // For desktop compat, we keep the exception semantics even though they are not ideal
+                // because a duplicate will raise exception causing certs to be removed even if 
+                // they weren't there before.
+                for (int j = 0; j < i; j++)
+                {
+                    Remove(certificates[j]);
```

Comment:
Where do we expect the exceptions come from?  I'm wondering if an exception getting thrown may represent some kind of corruption to the collection, such that then attempting to remove from it could be problematic.